tgbotapi.core/src/commonMain/kotlin/dev/inmo/tgbotapi/types/payments/stars/TransactionType.kt

```diff
@@ -0,0 +1,38 @@
+package dev.inmo.tgbotapi.types.payments.stars
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+
+@Serializable(with = TransactionTypeSerializer::class)
+enum class TransactionType {
```

Comment:
I think here must be string value in each transaction type representing its value in telegram

Reply:
I'd argue that this implementation is more general and intuitive. We would still need a custom serializer if I gave every enum value an `APIName` field or something like that.

Reply:
It is still must have the name of type same as in Bots API. Because of they are literally represent them